addons/account_peppol/static/src/components/res_config_settings_buttons/res_config_settings_buttons.js

```diff
@@ -91,9 +91,13 @@ class PeppolSettingsButtons extends Component {
         });
     }
 
-    deregister() {
+    async deregister() {
         if (this.ediMode === 'demo' || !['sender', 'smp_registration', 'receiver'].includes(this.proxyState)) {
-            this._callConfigMethod("button_deregister_peppol_participant");
+            await this._callConfigMethod("button_deregister_peppol_participant");
+            // Discard any changes
+            if (['not_registered', 'in_verification'].includes(this.proxyState)) {
```

Comment:
```suggestion
            if (['not_registered', 'in_verification'].includes(this.proxyState)) {
```
remove this condition